src/AaveV3StrategicAssets_20230606/StrategicAssetsManager.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.19;
+
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {Initializable} from 'solidity-utils/contracts/transparent-proxy/Initializable.sol';
+import {SafeERC20} from 'solidity-utils/contracts/oz-common/SafeERC20.sol';
+import {AaveGovernanceV2} from 'aave-address-book/AaveGovernanceV2.sol';
+import {AaveV3Ethereum} from 'aave-address-book/AaveV3Ethereum.sol';
+
+import {LSDLiquidityGaugeManager} from './LSDLiquidityGaugeManager.sol';
+import {VeTokenManager} from './VeTokenManager.sol';
+import {VlTokenManager} from './VlTokenManager.sol';
+
+contract StrategicAssetsManager is
+  Initializable,
+  LSDLiquidityGaugeManager,
+  VeTokenManager,
+  VlTokenManager
+{
+  using SafeERC20 for IERC20;
+
+  event WithdrawalERC20(address indexed _token, uint256 _amount);
+
+  function initialize() external initializer {
+    _transferOwnership(AaveGovernanceV2.SHORT_EXECUTOR);
+    _updateGuardian(_msgSender());
```

Comment:
@efecarranza this will set the guardian as the proxy factory contract which I think is not desired.

This is because when `create()` is called on the proxy factory, it will create a proxy contract for asset manager, then set the implementation address and then will call the `initialize()` method, so the `msg.sender` will be the proxy factory contract.

Reply:
hey @brotherlymite, yeah, should have added a comment that it was just a placeholder. there's an ARFC coming on who the guardian will be.